clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1282,6 +1282,11 @@ private Fetch<K, V> pollForFetches(Timer timer) {
         long pollTimeout = coordinator == null ? timer.remainingMs() :
                 Math.min(coordinator.timeToNextPoll(timer.currentTimeMs()), timer.remainingMs());
 
+        if(coordinator.isCommittingOffsetAsync()) {
```

Comment:
Hmm, do we want to do this in the general case? I think my expectation is that we would not continue fetching for partitions only when we have sent the offset commit and we are awaiting revocation as part of a rebalance.